contracts/core/extensions/CoreIssuanceOrder.sol

```diff
@@ -276,4 +292,96 @@ contract CoreIssuanceOrder is
+        uint toTaker = _requiredMakerTokenAmount.sub(_makerTokenUsed);
+
+        // Send left over maker token balance to taker
+        ITransferProxy(state.transferProxyAddress).transfer(
+            _order.makerToken,
+            toTaker,
+            _order.makerAddress,
+            msg.sender
+        );
+
+        // Calculate fees required
+        uint requiredFees = _order.relayerTokenAmount.mul(_fillQuantity).div(_order.quantity);
+
+        //Send fees to relayer
+        ITransferProxy(state.transferProxyAddress).transfer(
+            _order.relayerToken,
+            requiredFees,
+            _order.makerAddress,
+            _order.relayerAddress
+        );
+        ITransferProxy(state.transferProxyAddress).transfer(
+            _order.relayerToken,
+            requiredFees,
+            msg.sender,
+            _order.relayerAddress
+        );
+    }
+
+    function settleOrder(
+        OrderLibrary.IssuanceOrder _order,
+        uint _fillQuantity,
+        bytes _orderData
+    )
+        private
+    {
+        // Check to make sure open order amount equals _fillQuantity
+        uint closedOrderAmount = state.orderFills[_order.orderHash].add(state.orderCancels[_order.orderHash]);
+        uint openOrderAmount = _order.quantity.sub(closedOrderAmount);
+        require(
+            openOrderAmount >= _fillQuantity,
+            INVALID_FILL_AMOUNT
+        );
+
+        uint[] memory requiredBalances = new uint[](_order.requiredComponents.length);
+
+        // Calculate amount of maker token required
+        // Look into rounding errors
+        uint requiredMakerTokenAmount = _order.makerTokenAmount.mul(_fillQuantity).div(_order.quantity);
+
+        // Calculate amount of component tokens required to issue
+        for (uint16 i = 0; i < _order.requiredComponents.length; i++) {
+            // Get current vault balances
+            uint tokenBalance = IVault(state.vaultAddress).getOwnerBalance(
+                _order.makerAddress,
+                _order.requiredComponents[i]
+            );
+
+            // Amount of component tokens to be added to Vault
+            uint requiredAddition = _order.requiredComponentAmounts[i].mul(_fillQuantity).div(_order.quantity);
+
+            // Required vault balances after exchange order executed
+            requiredBalances[i] = tokenBalance.add(requiredAddition);
+        }
+
+        // Execute exchange orders
+        uint makerTokenAmountUsed = executeExchangeOrders(_orderData, _order.makerAddress);
+        require(makerTokenAmountUsed <= requiredMakerTokenAmount);
+
+        // Check that maker's component tokens in Vault have been incremented correctly
+        for (i = 0; i < _order.requiredComponents.length; i++) {
+            uint currentBal = IVault(state.vaultAddress).getOwnerBalance(
+                _order.makerAddress,
+                _order.requiredComponents[i]
+            );
+            //require(currentBal >= requiredBalances[i]);
```

Comment:
This is pending taker wallet exchange I assume?

Reply:
Si